src/actions/create-transak-session.ts

```diff
@@ -0,0 +1,50 @@
+type WidgetSessionBody = {
+  widgetParams: {
+    apiKey: string
+    referrerDomain: string
+    fiatAmount: string
+    defaultFiatAmount: string
+    defaultCryptoAmount: string
+    fiatCurrency: string
+    cryptoCurrencyCode: string
+    walletAddress: string
+  }
+}
+
+export const createTransakSession = async (
+  environment: string,
+  body: WidgetSessionBody, 
+  accessToken: string
+) => {
+  try {
+    const baseUrl =
+      environment === "production"
+        ? "https://api-gateway.transak.com"
+        : "https://api-gateway-stg.transak.com"
```

Comment:
Just a general thought. There's something about wrapping large blocks of code in a try/catch that I find really unclear. For example, there's no realistic way that this setting of a `baseUrl` is going to throw, so why is it within the try/catch? It all makes it harder to see exactly what bit of code we think might throw an error.